apps/dav/js/settings-admin-carddav.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * @copyright 2017, Bjoern Schiessle <bjoern@schiessle.org>
+ *
+ * @author Bjoern Schiessle <bjoern@schiessle.org>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+"use strict";
+
+document.getElementById('carddavExposeSystemAddressBook').onclick = function() {
+	OCP.AppConfig.setValue('dav', 'exposeSystemAddressBook', this.checked ? 'yes' : 'no');
+};
```

Comment:
I don't think it will work, we don't support inline js event.
Watch the change event?

```suggestion
document.getElementById('carddavExposeSystemAddressBook').addEventListener('change', function(e) {
	OCP.AppConfig.setValue('dav', 'exposeSystemAddressBook', e.target.checked ? 'yes' : 'no');
});
```

Reply:
During my tests it worked... But you know JS way better than I, so I'm happy to change it :-)